atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionTest.java

```diff
@@ -1208,6 +1213,35 @@ public void testIgnoresOrphanedSweepSentinel() {
         assertThat(txn.get(TABLE, ImmutableSet.of(cell)), is(ImmutableMap.of()));
     }
 
+    @Test
+    public void checkImmutableTsLockAfterReadsForConservativeIfFlagIsSet() {
+        TimelockService timelockService = spy(new LegacyTimelockService(timestampService, lockService, lockClient));
+        long transactionTs = timelockService.getFreshTimestamp();
+        LockImmutableTimestampResponse res =
+                timelockService.lockImmutableTimestamp();
+
+        setTransactionConfig(ImmutableTransactionConfig.builder()
+                .lockImmutableTsOnReadOnlyTransactions(true)
+                .build());
+
+        SnapshotTransaction transaction = getSnapshotTransactionWith(
+                timelockService,
+                () -> transactionTs,
+                res,
+                PreCommitConditions.NO_OP,
+                true);
+
+        transaction.get(TABLE_SWEPT_CONSERVATIVE, ImmutableSet.of(TEST_CELL));
+        transaction.commit();
+        timelockService.unlock(ImmutableSet.of(res.getLock()));
```

Comment:
`commit` currently checks these locks, so I'd expect two refreshes. Can we confirm the call explicitly after the `get` and `commit` calls?

Reply:
in this case we will check the lock only once after the get call - but not after the commit call. We either check immutableTsLock after each read (and don't check at commit) or only check at commit - depending on the value of `validateLocksOnReads`
I will move the verify after get() call.